### PR TITLE
Icmp plugin

### PR DESCRIPTION
**Pull request by BonnyAD9**

### CODE REVIEW

File: process/icmp.hpp [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+   }
+
+   std::string get_text() const
+   {
+      // type is on the first byte, code is on the second byte
+      auto *type_code = reinterpret_cast<const uint8_t *>(&this->type_code);
+
+      std::ostringstream out;
+      out << "type=\"" << (int)type_code[0] << '"'
+         << ",code=\"" << (int)type_code[1] << '"';
```
Comment (cejkato2) [inline process/icmp.hpp:124]:
```suggestion
      out << "type=\"" << (int) type_code[0] << '"'
         << ",code=\"" << (int) type_code[1] << '"';
```

File: process/icmp.cpp [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+ProcessPlugin *ICMPPlugin::copy()
+{
+   return new ICMPPlugin(*this);
+}
+
+int ICMPPlugin::post_create(Flow &rec, const Packet &pkt)
+{
+   if (pkt.ip_proto == IPPROTO_ICMP ||
+       pkt.ip_proto == IPPROTO_ICMPV6) {
+      if (pkt.payload_len < 2)
```
Comment (jaroslavpesek) [inline process/icmp.cpp:70]: for the sake of better readability I would avoid magic constants and use #define with sane name

**Review by jaroslavpesek — COMMENTED** [review]
It seems ok to me. @BonnyAD9, have you validated the results?

**Comment by BonnyAD9** [comment]
> It seems ok to me. @BonnyAD9, have you validated the results? If so, it is ready for merge, I believe.

Yes, it should work properly.

**Review by jaroslavpesek — APPROVED** [review]

**Pull request merged**